srv/salt/_modules/cephimages.py

```diff
@@ -6,20 +6,30 @@
 from glob import glob
 from subprocess import Popen, PIPE
 
+def _is_installed(name):
+    try:
+        devnull = open(os.devnull)
+        Popen([name], stdout=devnull, stderr=devnull).communicate()
+    except OSError as e:
+        if e.errno == os.errno.ENOENT:
+            return False
+    return True
```

Comment:
Does this actually execute the binary? That's fine for "rados" but if the binary does something like `rm -rf` (deliberate exaggeration) it might have "side effects". To find out if "foo" is installed and in the PATH, you can do "type foo" and examine the exit code:

```
vanguard1:/home/smithfarm # type rados
rados is hashed (/usr/bin/rados)
vanguard1:/home/smithfarm # echo $?
0
vanguard1:/home/smithfarm # type foo
bash: type: foo: not found
vanguard1:/home/smithfarm # echo $?
1
```

Anyway, this is "just an idea".

Reply:
cant you use zypper module or the pkg module in this case?

Reply:
Although this is neither a public API, nor a user specified variable you are probably right in terms of not taking the risk. `type/which` has the same effect and reduces the overall risk and maybe even the invocation speed. 

updating.